lightningd/log.c

```diff
@@ -569,28 +566,27 @@ static void log_dump_to_file(int fd, const struct log_book *lr)
 /* FIXME: Dump peer logs! */
 void log_backtrace_exit(void)
 {
+	int fd;
+	char logfile[sizeof("/tmp/lightning-crash.log.%u") + STR_MAX_CHARS(int)];
+
 	if (!crashlog)
 		return;
 
-	/* If we're not already pointing at a log file, make one */
-	if (crashlog->lr->print == log_to_stdout) {
-		const char *logfile = NULL;
-		int fd;
-
-		/* We expect to be in config dir. */
-		logfile = "crash.log";
-		fd = open(logfile, O_WRONLY|O_CREAT|O_APPEND, 0600);
-		if (fd < 0) {
-			logfile = "/tmp/lightning-crash.log";
-			fd = open(logfile, O_WRONLY|O_CREAT, 0600);
-		}
+	/* We expect to be in config dir. */
+	snprintf(logfile, sizeof(logfile), "crash.log.%u", getpid());
```

Comment:
I'm wondering if `pid` is the best identifier here, we might want to use something like `strftime(buffer, 26, "%Y%m%d%H%M%S", tm_info);` in order to easily identify the last crash or correlate a crash dump with external monitoring.